lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
I don't have data to back this up, but I think it is common on the web (with mix in, and monkey patching) that the prototype chain may have a fix of built in prototype version non-built in prototypes.

I am wondering whether this optimization is worth it at all to skip the any prototype if the prototype chain has any enumerable properties at all.  Can we measure how much that is going to help us?  Does it impact benchmark if we remove it?